### ISSUE TITLE
Feature Request: Add Support for Events

### ISSUE THREAD

**Issue by stefets** [4br3mm0rd/mpyg321#42]
## Feature Request: Add Support for Events

**Description:**
The current version of mpyg321 lacks support for events, which would be beneficial for enabling event-driven programming patterns.

**Reasoning:**
- Events provide a powerful mechanism for decoupling components and responding to state changes or actions. Introducing events into the library would enhance its usability and flexibility for various use cases.
- The wrapper has few callbacks and we must inherit it to override them; with this pattern, we can use the wrapper as composition instead of inheritance.

**Expected Behavior:**
- The library should include a mechanism for defining and subscribing to events.
- Users should be able to have subscribed handlers invoked accordingly.

**Additional Information:**
- Supporting events can lead to cleaner, more maintainable code by promoting separation of concerns and reducing direct dependencies between components.
- There will be **no breaking changes** if we decide to keep the public callbacks; indeed, it will be still possible to inherit for the callbacks and use the event pattern.
- The event API https://github.com/pyeve/events will be used for the refactoring

**Implementation Suggestions:**
Consider implementing an event system similar to the observer pattern, where objects can register event handlers and be notified when events occur.

**Examples:**
Here's an example of how event support could be used in the library, tested in my actual POC:
```python
from events import Events

events = Events()

# Emit an event
events.on_user_pause()

# Subscribe to an event
self.events.on_user_pause += self.handle_pause

def handle_pause(self):
    # Client logic
    pass

**Comment by stefets** [comment]
**References:**
- The event API project: https://github.com/pyeve/events
- WIP : https://github.com/stefets/mpyg321/tree/events

**Comment by 4br3mm0rd** [comment]
Hey! Nice to read from you!

It looks interesting, I'll give it a look. What about using decorators on the event callback instead of explicitly adding it?

**Comment by stefets** [comment]
> It looks interesting, I'll give it a look. What about using decorators on the event callback instead of explicitly adding it?

Hi! if a decorator can help simplify the modification that's perfect. See my last push for the implementation :)

Is that what you meant?

https://github.com/stefets/mpyg321/commit/10df26a572b80983d8b2431e2d561ddd0ba5c8ea

**Comment by 4br3mm0rd** [comment]
Hey!
Something like that. I just developed it the way I see it in PR #43 

As you can see, events can be subscribed to in 2 ways:
```
player = MPyg321Player()

@player.on("any_stop")
def my_callback():
  print("any stop")

def other_callback():
  print("any_stop")

player.subscribe_event("any_stop", other_callback)
```

I also removed the usage of Events, I don't think we need it.

let me know what you think about it :-)

**Comment by stefets** [comment]
Hi! 
If your modification can replace Events that's even better, it's one less dependency but I want to avoid reinventing the wheel and the code that Events allows must be managed in the mpyg321 lib now. I will extract your branch and test in my configuration, I will come back to you in a few days with my results and suggestions! 
Thank you very much !

**Comment by 4br3mm0rd** [comment]
Released as 2.2.0
